diff --git a/straight.el b/straight.el
index 0437ab4..754fd88 100644
--- a/straight.el
+++ b/straight.el
@@ -1702,6 +1702,12 @@ edit. Otherwise, PROMPT and ACTIONS are as for
              (or straight--default-directory default-directory))
         (straight--recursive-edit)))))
 
+(defun get-mirror-host (site default-host) 
+   (let ((mirror (getenv (upcase (format "%s_mirror" site)))))
+      (if mirror
+          mirror 
+          default-host)))
+
 (defun straight-vc-git--encode-url (repo host &optional protocol)
   "Generate a URL from a REPO depending on the value of HOST and PROTOCOL.
 REPO is a string which is either a URL or something of the form
@@ -1719,6 +1725,7 @@ See also `straight-vc-git--decode-url'."
     ((or `github `gitlab `bitbucket)
      (let ((domain (pcase host
                      (`bitbucket "bitbucket.org")
+		     (`github (get-mirror-host "github" "github.com"))
                      (_ (format "%s.com" host)))))
        (pcase (or protocol straight-vc-git-default-protocol)
          (`https
@@ -4891,7 +4898,7 @@ action, just return it)."
          (recipe (straight--convert-recipe melpa-recipe)))
     (straight--with-plist recipe (host repo)
       (pcase host
-        (`github (browse-url (format "https://github.com/%s" repo)))
+        (`github (browse-url (format "https://%s/%s" (get-mirror-host "github" "github.com") repo)))
         (`gitlab (browse-url (format "https://gitlab.com/%s" repo)))
         (_ (browse-url (format "%s" repo)))))))
 
