diff --git a/eglot.el b/eglot.el
index db468d8..08a43f2 100644
--- a/eglot.el
+++ b/eglot.el
@@ -1468,6 +1468,12 @@ For example, to keep your Company customization use
   "A hook run by EGLOT after it started/stopped managing a buffer.
 Use `eglot-managed-p' to determine if current buffer is managed.")
 
+(defvar eglot-disable-flymake nil
+  "disable flymake in eglot")
+
+(defvar eglot-disable-eldoc nil
+  "disable eldoc in eglot")
+
 (define-minor-mode eglot--managed-mode
   "Mode for source buffers managed by some EGLOT project."
   :init-value nil :lighter nil :keymap eglot-mode-map
@@ -1500,8 +1506,10 @@ Use `eglot-managed-p' to determine if current buffer is managed.")
     (unless (eglot--stay-out-of-p 'imenu)
       (add-function :before-until (local 'imenu-create-index-function)
                     #'eglot-imenu))
-    (flymake-mode 1)
-    (eldoc-mode 1)
+    (unless eglot-disable-flymake
+      (flymake-mode 1))
+    (unless eglot-disable-eldoc
+      (eldoc-mode 1))
     (cl-pushnew (current-buffer) (eglot--managed-buffers (eglot-current-server))))
    (t
     (remove-hook 'after-change-functions 'eglot--after-change t)
